src/main/java/com/booking/superride/repository/RideRepository.java

```diff
@@ -0,0 +1,21 @@
+package com.booking.superride.repository;
+
+import com.booking.superride.entity.RideDetails;
+import com.booking.superride.entity.projections.AvailableTaxiFareDetails;
+import com.booking.superride.entity.projections.OnTripTaxiDetails;
+import org.springframework.cglib.core.Local;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+
+public interface RideRepository extends JpaRepository<RideDetails, Long> {
+
+    List<OnTripTaxiDetails> findAllByRideDateAndDropTimeGreaterThanEqual(LocalDate date, LocalDateTime dateTime);
+
+    @Query(value = "SELECT taxiid as TaxiId, sum(fare) as fare from ride_details where ridedate = :rideDate and taxiid in :taxiids group by taxiid", nativeQuery = true)

```

Comment:
if possible, use JPQL. if there are any changes around table name then we need to make change here as well.